services/gatekeeper/server/src/gatekeeper-api.js

```diff
@@ -176,69 +176,162 @@ v1router.get('/status', async (req, res) => {
+ *                         description: When the signature was created.
+ *                       signer:
+ *                         type: string
+ *                         description: The DID of the signer (often the same as `did`).
+ *                       hash:
+ *                         type: string
+ *                         description: Optional hash of the operation payload.
+ *
+ *               - type: object
+ *                 required: [ type, did, signature ]
+ *                 properties:
+ *                   type:
+ *                     type: string
+ *                     enum: [ "delete" ]
+ *                     description: Must be "delete" to deactivate an existing DID.
+ *                   did:
+ *                     type: string
+ *                     description: The DID to deactivate.
+ *                   signature:
+ *                     type: object
+ *                     required: [ value, signed ]
+ *                     description: Cryptographic signature verifying this delete operation.
+ *                     properties:
+ *                       value:
+ *                         type: string
+ *                         description: The signature value (base64, hex, etc.).
+ *                       signed:
+ *                         type: string
+ *                         format: date-time
+ *                         description: When the signature was created.
+ *                       signer:
+ *                         type: string
+ *                         description: The DID of the signer, who must have authority to delete.
+ *                       hash:
+ *                         type: string
+ *                         description: Optional hash of the operation payload.
+ *
  *     responses:
  *       200:
- *         description: Successfully created the DID.
- *       400:
- *         description: Invalid request.
+ *         description: >
+ *           - If `type = "create"`, returns the newly created DID as a string.
+ *           - Otherwise (for update or delete), returns a value indicating success.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               oneOf:
+ *                 - type: string
+ *                   description: A DID string (when a create operation succeeds).
+ *                 - type: boolean
+ *                   description: A success indicator for update/delete operations.
  *       500:
  *         description: Internal Server Error.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: string
  */
 v1router.post('/did', async (req, res) => {
     try {
         const operation = req.body;
-        const did = await gatekeeper.createDID(operation);
+        let did;
+        if (operation && operation.type === "create") {
+            did = await gatekeeper.createDID(operation);
+        } else {
+            did = await gatekeeper.updateDID(operation);
```

Comment:
Maybe rename local var to something more generic given that updateDID returns a boolean success flag?